File: 2231-largest-number-after-digit-swaps-by-parity/2231-largest-number-after-digit-swaps-by-parity.java
class Solution {
    public int largestInteger(int num) {
        PriorityQueue<Integer> even = new PriorityQueue<>( (a,b) -> b-a);
        PriorityQueue<Integer> odd = new PriorityQueue<>( (a,b) -> b-a);
        String n = num + "";
        while(num>0) {
            int digit = num % 10;
            if(digit%2==0)
                even.add(digit);
            else
                odd.add(digit);
            num = num / 10;
        }
        StringBuilder res = new StringBuilder();
        for(int i=0;i<n.length();i++) {
            int val = Integer.parseInt(n.charAt(i) + "");
            if(val%2==0) {
                res.append(even.remove());
            }
            else {
                res.append(odd.remove());
            }
        }
        return Integer.parseInt(res.toString());
    }
}

File: 2231-largest-number-after-digit-swaps-by-parity/2231-largest-number-after-digit-swaps-by-parity.cpp
class Solution {
public:
    int largestInteger(int num) {
    int cnt[10] = {}, p[2] = {0, 1}, res = 0;
    for (int n = num; n > 0; n /= 10)
        ++cnt[n % 10];
    for (long long n = num, mul = 1; n > 0; n /= 10, mul *= 10) {
        int par = n % 10 % 2 == 1;
        while (cnt[p[par]] == 0)
            p[par] += 2;
        res += mul * p[par];
        --cnt[p[par]];
    }
    return res;
}
};